lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -448,23 +449,43 @@ def normalize_allocation_exclusions(exclusions: Union[List[str], Set[str], str])
         return exclusions
 
     @staticmethod
-    def missing_sys_requirements() -> List[str]:
+    def running_as_lxc():
+        """Check if the current process is running inside an LXC container.
+
+        TODO: if canonical/operator-libs-linux#127 fixes, we may remove this method.
+        """
+        try:
+            return subprocess.run(["systemd-detect-virt", "--container"]).returncode == 0
+        except FileNotFoundError:
+            # No systemd! Either this is a docker container OR a very old distro
+            return False
+
+    @staticmethod
+    def missing_sys_requirements(name: str) -> List[str]:
         """Checks the system requirements."""
         missing_requirements = []
 
-        max_map_count = int(subprocess.getoutput("sysctl vm.max_map_count").split("=")[-1].strip())
-        if max_map_count < 262144:
-            missing_requirements.append("vm.max_map_count should be at least 262144")
+        config = {
+            "vm.max_map_count": "262144",
+            "vm.swappiness": "0",
+            "net.ipv4.tcp_retries2": "5",
+        }
+        if OpenSearchDistribution.running_as_lxc():
```

Comment:
I think it is better to check sysctl anyway, it doesn't matter on which platform.

Reply:
So, I cannot apply all the options above for LXC; but I can if we have a VM or host.

Reply:
I think it is better to check the sysctl on all platforms after setting. It doesn't matter what platform or exit codes are received.

Reply:
@delgod, so to clarify, it is better to keep the original behavior, where we check the settings and then, block if the user did not apply it?

Reply:
@phvalguima I think the following flow can work in our case:
1. check the current values of each sysctl setting (regardless of the platform)
2. make the distinction between: 
   - the settings that need to be `equal` to a value (`tcp_retries2`, or `swappiness`) 
   - the settings that need to be `greater` than a value (`max_map_count >= 262144` , file-max)
       - we should only set these if that's the case, otherwise we are decreasing the quality of the settings already existing in the environment
3. This should result in a set of `invalid` settings.
4. attempt to set these settings (all the invalid set or one at a time), and re-check the new values, so you can accurately add each in the missing_requirements if any:
   - I prefer the missing_requirements we had, as the message is more tailored towards the values that need to be set, not just that the setting could not be set